core/arch/arm/plat-rockchip/platform_rk3588.c

```diff
@@ -418,6 +421,9 @@ TEE_Result tee_otp_get_hw_unique_key(struct tee_hw_unique_key *hwkey)
 
 	mutex_lock(&huk_mutex);
 
+	if (huk)
+		goto cached;
+
 	res = tee_rockchip_read_huk(hwkey);
```

Comment:
It would look simpler to allocate `huk` here, then generate the HUK in `huk` and copy it to `hwkey`  at the end, upon success.

Reply:
If the `huk` is allocated first, it's more difficult to handle errors when reading or generating the HUK fails. A second call to `tee_otp_get_hw_unique_key` would see that `huk` is set and use the cached, but invalid value. Thus, the error handling would have to ensure that `huk` is explicitly freed and set to `NULL` in case of an error. On the other hand, explicit error handling would make this more clear. I'll give it a try.